src/lib/OpenEXRCore/internal_zstd.c

```diff
@@ -0,0 +1,115 @@
+/*
+** SPDX-License-Identifier: BSD-3-Clause
+** Copyright Contributors to the OpenEXR Project.
+*/
+
+#include <openexr_compression.h>
+#include "internal_compress.h"
+#include "internal_decompress.h"
+#include "blosc2.h"
+
+size_t
+exr_get_zstd_lines_per_chunk ()
+{
+    return 1;
+}
+
+long
+exr_compress_zstd (char* inPtr, int inSize, void* outPtr, int outPtrSize)
+{
+    if (inSize == 0) // Weird input data when subsampling
+    {
+        outPtr = NULL;
+        return 0;
+    }
+
+    blosc2_cparams cparams = BLOSC2_CPARAMS_DEFAULTS;
+    int typeSize     = inSize % 4 == 0 ? 4 : 2;
+    cparams.typesize = typeSize;
+    // clevel 9 is about a 20% increase in compression compared to 5.
+    // Decompression speed is unchanged.
+    int zstd_level;
+    exr_get_default_zstd_compression_level (&zstd_level);
```

Comment:
Should this be getting the current compression level, not the default one? Changing the compression level doesn't seem to make a difference to the output size